modin/engines/base/io/column_stores/parquet_dispatcher.py

```diff
@@ -79,58 +78,22 @@ def _read(cls, path, engine, columns, **kwargs):
                 return cls.single_worker_read(
                     path, engine=engine, columns=columns, **kwargs
                 )
-        else:
-            directory = False
-        if not columns:
-            import s3fs
-
-            if directory:
-                # Path of the sample file that we will read to get the remaining columns
-                pd = ParquetDataset(path)
-                meta = pd.metadata
-                column_names = pd.schema.to_arrow_schema().names
-            elif isinstance(path, str) and path.startswith("hdfs://"):
-                import fsspec.core
 
-                fs, path = fsspec.core.url_to_fs(path)
-                pd = ParquetDataset(path, filesystem=fs)
-                meta = pd.metadata
-                column_names = pd.schema.to_arrow_schema().names
-            elif isinstance(path, s3fs.S3File) or (
-                isinstance(path, str) and path.startswith("s3://")
-            ):
-                from botocore.exceptions import NoCredentialsError
+        if not columns:
+            import fsspec.core
+            from pandas.io.common import is_fsspec_url
 
-                if isinstance(path, s3fs.S3File):
-                    bucket_path = path.url().split(".s3.amazonaws.com")
-                    path = "s3://" + bucket_path[0].split("://")[1] + bucket_path[1]
-                try:
-                    fs = s3fs.S3FileSystem()
-                    pd = ParquetDataset(path, filesystem=fs)
-                except NoCredentialsError:
-                    fs = s3fs.S3FileSystem(anon=True)
-                    pd = ParquetDataset(path, filesystem=fs)
-                meta = pd.metadata
-                column_names = pd.schema.to_arrow_schema().names
-            else:
-                meta = ParquetFile(path).metadata
-                column_names = meta.schema.to_arrow_schema().names
+            fs, path_ = (
+                fsspec.core.url_to_fs(path, **(kwargs["storage_options"] or {}))
```

Comment:
```suggestion
                fsspec.core.url_to_fs(path, **(kwargs.get("storage_options") or {}))
```

Reply:
What's benefit in this case?

Reply:
Everything is the same, except that no exceptions will be thrown if `storage_options` key is not present.